t/Util.pm

```diff
@@ -820,4 +824,65 @@ sub spawn_dns_server {
     return $server;
 }
 
+sub run_openssl_client {
+    my($opts) = @_;
+    my $port = $opts->{port} or croak("`port` is required!");
+    my $san = $opts->{san};
+    my $host = $opts->{host} // '127.0.0.1';
+    my $path = $opts->{path} // '/';
+    my $timeout = $opts->{timeout} // 2.0;
+    my $request = $opts->{request};
+    my $request_default = $opts->{request_default} // 0;
+    my $ossl_opts = $opts->{opts} // '';
+    my $ossl_cmd = $opts->{ossl_cmd} // 'openssl';
+
+    my $tempdir = tempdir();
+    my $cmd = "$ossl_cmd s_client $ossl_opts -connect $host:$port";
+    if (defined $san && $san ne '') {
+        $cmd = $cmd." -servername $san";
+    }
+    diag("run_openssl_client: $cmd");
+    my $cpid = open3(my $chld_in, my $chld_out, my $chld_err = gensym, $cmd);
+    sleep ${timeout};
+    $chld_in->autoflush(1);
+    $chld_out->autoflush(1);
+    $chld_err->autoflush(1);
+    if ($request_default) {
+        $chld_in->eof() || print $chld_in <<"EOT";
+GET $path HTTP/1.1\r
+Host: $san:$port\r
+Connection: close\r
+\r
+EOT
+    } elsif (defined $request && $request ne '') {
+        $chld_in->eof() || print $chld_in "$request";
+    }
+
+    while ($timeout > 0.0) {
+        my $cpid_wait = waitpid($cpid, POSIX::WNOHANG);
+        if ($cpid_wait == $cpid || $cpid_wait == -1) {
+            last;
+        }
+
+        sleep 0.1;
+        $timeout -= 0.1
+    }
+
+    close $chld_in;
+    my $resp_out = do { local $/; <$chld_out> };
+    my $resp_err = do { local $/; <$chld_err> };
+    close $chld_out;
+    close $chld_err;
+
+    if ($timeout <= 0.0) {
+        kill 'KILL', $cpid;
+    }
+
+    return ($resp_out, $resp_err);
+}
+
+sub run_openssl_client_joined {
+  return join("\n", run_openssl_client(@_));
+}
```

Comment:
Do we need to define both `run_openssl_client` and `run_openssl_client_joined`? I ask this because I do not see former being called. Also, "joined" sounds a bit odd to me - it does not parse well unless to those who understand what the function does.

Would it make sense to return the joined string by default from `run_openssl_client`, while (potentially) adding a option to return an array instead?

Reply:
This has been incorporated in https://github.com/h2o/h2o/pull/2947/commits/ff454017b1adf6f9eab8d7caf7c5015df786db68.  I wasn't sure what to name the option, I used `split_return`, please let me know if you think I should change it.